### PR TITLE
Generalized plane strain

### PR DESCRIPTION
**Pull request by bwspenc**
refs #5042

### CODE REVIEW

File: modules/solid_mechanics/include/materials/PlaneStrain.h [hunk 1]
```diff
@@ -8,11 +8,14 @@
...
 #include "Element.h"
+#include "ScalarCoupleable.h"
 
 namespace SolidMechanics
 {
 
-class PlaneStrain : public Element
+class PlaneStrain :
+  public Element,
+  ScalarCoupleable
```
Comment (permcody) [inline modules/solid_mechanics/include/materials/PlaneStrain.h]: private inheritance, is that fine with you?

File: modules/solid_mechanics/src/kernels/OutOfPlaneStress.C [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Real
+OutOfPlaneStress::computeQpJacobian()
+{
+  Real C33 = _Jacobian_mult[_qp].valueAtIndex(11);
+  return C33*_test[_i][_qp]*_phi[_j][_qp];
+}
+
+Real
+OutOfPlaneStress::computeQpOffDiagJacobian(unsigned int jvar)
```
Comment (jwpeterson) [inline modules/solid_mechanics/src/kernels/OutOfPlaneStress.C]: `jvar` will trigger an unused variable warning in dbg mode.

File: modules/solid_mechanics/include/materials/PlaneStrain.h (hunk 1 above)
Comment (bwspenc) [inline modules/solid_mechanics/include/materials/PlaneStrain.h]: I didn't actually intend to privately inherit ScalarCoupleable -- it sounds like you have to list public before each class you inherit from for them to be public.  It probably doesn't make much difference, but I'll change it to be public.

File: modules/solid_mechanics/src/kernels/OutOfPlaneStress.C (hunk 2 above)
Comment (bwspenc) [inline modules/solid_mechanics/src/kernels/OutOfPlaneStress.C]: Good catch -- I'll fix that.  This method is a placeholder for when I actually implement it.

**Pull request merged**